### ISSUE TITLE
filteredStream.SearchStream: unexpected end of JSON input

### ISSUE THREAD

**Issue by mfreeman612** [michimani/gotwi#259]
I'm having an issue when trying to Search a stream:

```
func ListenTweets(c *gotwi.Client) error {

	fmt.Println("Listening for Tweets...")
	p := &types.SearchStreamInput{}

	s, err := filteredstream.SearchStream(context.Background(), c, p)
	if err != nil {
		return err
	}

	for s.Receive() {
		fmt.Println("Received")

		t, err := s.Read()
		if err != nil {
			fmt.Println("Error:", err)
			s.Stop()
			break
		} else {
			fmt.Println(gotwi.StringValue(t.Data.ID), gotwi.StringValue(t.Data.Text))
		}
	}
	return nil
}
```

When I run this, sometimes it works, but usually it fails with the "unexpected end of JSON input" error.

Please advice

**Comment by michimani** [comment]
@mfreeman612  

Thank you for reporting the issue.

I don't know, since I haven't actually run it, but I suspect that the JSON parsing is failing because `t` is an empty string in the following location.

https://github.com/michimani/gotwi/blob/1bcd4685da34df4ac0304e47114a8316ce9be7d5/stream_client.go#L54

So it looks like you need to check for a character from `t` in the above section. This is a problem on this library side and we will fix it as soon as possible.

Sorry for the inconvenience.

**Comment by michimani** [comment]
Thanks for creating PR! I will check it.

**Comment by michimani** [comment]
@mfreeman612 

Earlier, I merged your PR that resolves this issue into the main branch and released v0.11.6.

Thank you for your contribution!
